includes/class-amp-story-post-type.php

```diff
@@ -1553,4 +1589,222 @@ public static function add_new_max_image_size( $image_sizes ) {
 
 		return $image_sizes;
 	}
+
+	/**
+	 * Checks for `story_export` and valid Ajax nonce.
+	 *
+	 * @return bool
+	 */
+	public static function can_export() {
+		return is_singular( self::POST_TYPE_SLUG ) && isset( $_GET['story_export'] ) && check_ajax_referer( self::AMP_STORIES_AJAX_ACTION, 'nonce', false );
+	}
+
+	/**
+	 * Get the args used during a story export.
+	 *
+	 * @param string $slug The slug used to build the new `canonical_url`.
+	 *
+	 * @return array
+	 */
+	public static function get_export_args( $slug = '' ) {
+		$base_url = untrailingslashit( AMP_Options_Manager::get_option( 'story_export_base_url' ) );
+
+		return [
+			'base_url'      => esc_url( $base_url ),
+			'canonical_url' => ( $base_url && $slug ) ? esc_url( trailingslashit( $base_url ) . $slug ) : false,
+		];
+	}
+
+	/**
+	 * Returns an asset basename where the date directory structure is retained to avoid filename collisions.
+	 *
+	 * This means that an asset like `https://sample.org/wp-content/uploads/2019/07/sample.jpg`
+	 * returns the basename `2019-07-sample.jpg` instead of `sample.jpg`.
+	 *
+	 * @param string $asset The URL of the export asset.
+	 *
+	 * @return string
+	 */
+	public static function export_image_basename( $asset ) {
+		$asset = preg_replace_callback(
+			'/uploads\/(.*)/',
+			function( $matches ) {
+				return str_replace( '/', '-', $matches[1] );
+			},
+			$asset
+		);
+
+		return basename( $asset );
+	}
+
+	/**
+	 * Ajax handler to export the story ZIP archive.
+	 *
+	 * This method returns an error as JSON and the binary data on success.
+	 */
+	public static function handle_export() {
+		check_ajax_referer( self::AMP_STORIES_AJAX_ACTION, 'nonce' );
+
+		// The user must have the correct permissions.
+		if ( ! current_user_can( 'edit_files' ) ) {
```

Comment:
This would allow export for only admins, probably editors and authors should also be able to export Stories?

Also, it looks like `edit_files` is not actually a used capability anymore, it's assigned to the admins but doesn't seem to be used for anything.

Looks like the tmp folder is used for generating the files before downloading, perhaps we should remove this condition and allow export for anyone who can edit and create AMP Stories. Thoughts?

cc @swissspidy @westonruter

Reply:
> allow export for anyone who can edit and create AMP Stories

Agreed. If the current user can edit the current story, they should be able to export it.

Reply:
It's been updated to `publish_posts`.